enhancements/kueue/openshift-kueue.md

```diff
@@ -0,0 +1,760 @@
+
+We will use the kueue configuration [API](https://github.com/kubernetes-sigs/kueue/blob/main/apis/config/v1beta1/configuration_types.go).
+
+The operator reads the kueue configuration and generates a ConfigMap that the kueue manager deployment will use. 
+
+The API is displayed below but we have requested an api-review [here](https://github.com/openshift/api/pull/2222).
+
+We have performed an API review in two stages.
+
+Stage 1 - https://github.com/openshift/api/pull/2222
+
+Stage 1 added Integrations as a required field for the operator.
+
+Stage 2 - https://github.com/openshift/api/pull/2250
+Stage 2 added support for ManagedJobWithoutQueueName, Fairsharing, and WaitForPodsReady.
+
+The API is defined in those PRs and in our operator.
+We will highlight how it can used in this enhancement rather than list out the API and distract the reader with code.
+We will use the kueue configuration [API](https://github.com/kubernetes-sigs/kueue/blob/main/apis/config/v1beta1/configuration_types.go).
+
+The operator reads the kueue configuration and generates a ConfigMap that the kueue manager deployment will use. 
+We will only expose a limited amount of configurations.
+
+The flow is as follows:
+
+1) User specifies a Kueue CRD
+1) Configuration is specified
+1) A Configmap will be created with the configuration specified
+1) Kueue deployment will use that configmap in the operand deployment.
+
+The operator will manage these components necessary for Kueue. 
+- Custom Resources
+- Cluster Roles
+- Cluster Role Bindings
+- Services
+- Mutating Webhook
+- Validating Webhook
+- ServiceMonitors
+- APIService for visibility
+- Roles
+- Certificates
+- Issuers
+
+The operator will create and apply these resources to the cluster.
+
+The operator has the following requirements.
+
+- Konflux integration
+- Disconnected
+- FIPS
+
+The operator will be OLM managed and hosted on OperatorHub.
+
+### Feature Support
+
+### Workflow Description
+
+#### Batch Job Administrator
+
+As a admin, I want to enable Kueue to manage batch jobs.
+
+```yaml
+apiVersion: operator.openshift.io/v1alpha1
+kind: Kueue
+metadata:
+ name: cluster
+ namespace: openshift-kueue-operator
+spec:
+   config:
+     integrations:
+       frameworks:
+         - BatchJob
+```
+
+This will create a Kueue deployment that will manage batch jobs. 
+
+#### Core Openshift Management
+
+```yaml
+apiVersion: operator.openshift.io/v1alpha1
+kind: Kueue
+metadata:
+  labels:
+    app.kubernetes.io/name: kueue-operator
+    app.kubernetes.io/managed-by: kustomize
+  name: cluster
+  namespace: openshift-kueue-operator
+spec:
+  managementState: Managed
+  config:
+    integrations:
+      frameworks:
+      - BatchJob
+      - Pod
+      - Deployment
+      - StatefulSet 
+```
+
+This can be used for Kueue to manage all workloads.
+A popular request is for Kueue to manage the access of GPUs for Model Serving.
+
+#### RHOAI Enablement
```

Comment:
Perhaps change RHOIA Enablement and IBM Enablement to example configurations, since they may change over time.

Reply:
done.